hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java

```diff
@@ -86,19 +91,25 @@ public SyntheticTxnFactory() {
 						.setToken(nftExchange.getTokenType())
 						.addNftTransfers(nftExchange.nftTransfer()));
 			}
+			final var tokenTransferListBuilder = TokenTransferList.newBuilder();
```

Comment:
Does this mean that each `TokenTransferWrapper` can have at most one `fungibleTransfer`?

Reply:
This should be declared inside the for loop and all fungibleTransfers of the TokenTransferWrapper should be added to the transaction body builder. It is fixed.